core/src/main/scala/org/apache/spark/scheduler/DAGScheduler.scala

```diff
@@ -1281,10 +1281,24 @@ class DAGScheduler(
         val failedStage = stageIdToStage(task.stageId)
         val mapStage = shuffleIdToMapStage(shuffleId)
 
+        def invalidateLostFilesAndExecutor(): Unit = {
+          // Mark the map whose fetch failed as broken in the map stage
+          if (mapId != -1) {
+            mapStage.removeOutputLoc(mapId, bmAddress)
+            mapOutputTracker.unregisterMapOutput(shuffleId, mapId, bmAddress)
+          }
+
+          // TODO: mark the executor as failed only if there were lots of fetch failures on it
+          if (bmAddress != null) {
+            handleExecutorLost(bmAddress.executorId, filesLost = true, Some(task.epoch))
```

Comment:
Ah sorry I concurrently edited my comments when I looked at this more.  I'm still confused though -- why does (7) matter (that the metadata for the executor didn't get cleaned up) given that it didn't have any shuffle outputs?